lib/std/ioutils.nim

```diff
@@ -0,0 +1,50 @@
+when defined(windows):
+  proc c_dup(oldfd: FileHandle): FileHandle {.
+    importc:"_dup", header: "<io.h>".}
+  proc c_dup2(oldfd: FileHandle, newfd: FileHandle): cint {.
+    importc: "_dup2", header: "<io.h>".}
+else:
+  proc c_dup(oldfd: FileHandle): FileHandle{.
+    importc: "dup", header: "<unistd.h>".}
+  proc c_dup2(oldfd: FileHandle, newfd: FileHandle): cint {.
+    importc: "dup2", header: "<unistd.h>".}
+
+when false:
+  const SupportIoctlInheritCtl = (defined(linux) or defined(bsd)) and
+                                not defined(nimscript)
+  when SupportIoctlInheritCtl:
+    var
+      FIOCLEX {.importc, header: "<sys/ioctl.h>".}: cint
+      FIONCLEX {.importc, header: "<sys/ioctl.h>".}: cint
+
+## Also defined in std/posix and system/io
+proc strerror(errnum: cint): cstring {.importc, header: "<string.h>".}
+when not defined(NimScript):
+  var errno {.importc, header: "<errno.h>".}: cint ## error variable
+
+template checkError(ret: cint)=
+  if ret == -1:
+    when not defined(NimScript):
+      raise newException(IOError, $strerror(errno))
+    else:
+      quit(1)
+
+proc duplicate*(oldfd: FileHandle): FileHandle =
+  ##[
+  Return a copy of the file handle `oldfd`.
+  After a successful return, both `FileHandle` may be used interchangeably.
+  They refer to the same open file description and share file offset and status flags.
+  Calls POSIX function `dup` on Linux and  `_dup` on Windows
+  ]##
+  let retValue = c_dup(oldfd)
+  checkError(retValue)
+  retValue
+
+proc duplicateTo*(oldfd: FileHandle, newfd: FileHandle) =
+  ##[
+  Perform the same task a `duplicate` but instead of using the lowest unused file descriptor
+  it uses the FileHandle` specified by `newfd`.
+  Calls POSIX function `dup2` on Linux and `_dup2` on Windows.
+  ]##
```

Comment:
can you add a very simple 2 or 3 line `runnableExamples:` to illustrate usage un-ambiguously (especially wrt argument ordering which is easy to forget / get wrong).

If not, then forget it but it would be nice. Ditto with `duplicate`.

Most new procs have `runnableExamples:` and it really improves docs.